### PR TITLE
feat(utils): Add the isWeekend method

### PR DESCRIPTION
**Pull request by bugtype**

### CODE REVIEW

File: src/plugins/withDateProps.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import withDateProps from './withDateProps'
 
 describe('withDateProps plugin', () => {
-  it('return with corret date, isCurrentDate false, isCurrentMonth true', () => {
+  it('return with correct date, isCurrentDate false, isCurrentMonth true', () => {
```
Comment (bugtype) [inline src/plugins/withDateProps.test.ts:4]: Recommend https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Comment by JaeYeopHan** [comment]
so sad

image

File: src/utils/isWeekend.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function isWeekend(targetDate: Date) {
```
Comment (JaeYeopHan) [inline src/utils/isWeekend.ts:1]: why don't use `isWeekend` in date-fns?

https://date-fns.org/v2.21.3/docs/isWeekend

Comment (bugtype) [reply src/utils/isWeekend.ts:3]: Reflected it.

Comment (JaeYeopHan) [reply src/utils/isWeekend.ts:3]: we don't need to this function! :)

File: src/plugins/withDateProps.ts [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+  isSameYearAndMonth,
+  isWeekend as _isWeekend,
+} from '../utils'
 
 export default function withDateProps(baseDate: Date, cursorDate: Date) {
   return function <T extends DateCell>(cell: T) {
     const { value: targetDate } = cell
     const isCurrentMonth = isSameYearAndMonth(cursorDate, targetDate)
     const isCurrentDate = isSameDate(baseDate, targetDate)
+    const isWeekend = _isWeekend(targetDate)
```
Comment (JaeYeopHan) [inline src/plugins/withDateProps.ts:15]: just use `isWeekend` function in date-fns !

from. https://github.com/veccu/react-calendar/pull/38/files#r633473934

File: src/utils/isWeekend.test.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import isWeekend from './isWeekend'
```
Comment (JaeYeopHan) [inline src/utils/isWeekend.test.ts:1]: we don't need to test date-fns function :)

**Comment by bugtype** [comment]
All right. :)

**Comment by JaeYeopHan** [comment]
> All right. :)

but `isWeekend` date props looks good to me! :)

**Comment by bugtype** [comment]
I'm just thinking, I have a few questions.

#### 1. How about handling the date logic in one place? (e.g. to create dependencies only in `utils` or `utils/date`.)

In the past, I've worked on changing `date-fns` while using `moment` in a project. 
There was a lot of `moment` legacy code in components, hooks, etc. It took me a long time to **refactor** this.
If we use it as we do now, in the distant future... we have to do a lot of work when we use a date library other than `date-fns`. (On the other hand, I think...  it could be overengineering. haha... )

#### 2. Why don't you provide a `format` method?

I think users should be able to configure the basic calendar **without** installing a separate date library? 
there may be a dependency conflict between the project and this library.
When a developer implements a calendar in a project, **the dependency conflict** may occur between the project and this library. if `date-fns` is used. In addition, _developers should **always** use a date library to indicate the date._ So, if we provide, this problem will not arise.

**Comment by JaeYeopHan** [comment]
thanks for your comment @bugtype :)

> 1. How about handling the date logic in one place? (e.g. to create dependencies only in utils or utils/date.)

good idea, but if I migrate to another library, I won't do it by hand, so I don't think this is a big problem. (maybe with jscodeshift tool)

> 2. Why don't you provide a format method?

maybe conflict is not problem. date-fns is [peerDeps](https://github.com/veccu/react-calendar/blob/main/package.json#L126)

but very good idea!! in the future, I **will** internalize date-fns functions used in the project!

**Pull request closed without merging**